frontend/src/app/projects/models/diagrams/model-diagram-dialog/model-diagram-dialog.component.ts

```diff
@@ -55,20 +55,19 @@ export class ModelDiagramDialogComponent {
 
   constructor(
     private modelDiagramService: ModelDiagramService,
-    private dialogRef: MatDialogRef<ModelDiagramDialogComponent>,
     private dialog: MatDialog,
     @Inject(MAT_DIALOG_DATA)
-    public data: { modelSlug: string; projectSlug: string },
+    public data: { model: Model; project: Project },
   ) {
     this.modelDiagramService
-      .getDiagramMetadata(this.data.projectSlug, this.data.modelSlug)
+      .getDiagramMetadata(this.data.project.slug, this.data.model.slug)
       .subscribe({
         next: (diagramMetadata) => {
           this.diagramMetadata = diagramMetadata;
           this.observeVisibleDiagrams();
         },
-        error: () => {
-          this.dialogRef.close();
+        error: (err) => {
+          this.errorMessage = err.error;
         },
```

Comment:
Please remove this and add `this.dialogRef.close()` back

Reply:
Done